src/main.js

```diff
@@ -11,6 +12,128 @@ let quotes = [];
 let categories = [];
 
 let lastDisplayedQuoteId = null; // Track last displayed quote's ID instead of index
+let currentQuoteId = null; // Track currently displayed quote ID for favorites
+
+// Check if Supabase is configured - declare early to avoid hoisting issues
+const hasSupabaseConfig = import.meta.env.VITE_SUPABASE_URL && import.meta.env.VITE_SUPABASE_ANON_KEY;
+let useMockData = !hasSupabaseConfig; // Use mock data if Supabase not configured
+
+if (useMockData) {
+  console.log("🎭 Running in MOCK MODE - No Supabase configuration found");
+  console.log("To use real database, add VITE_SUPABASE_URL and VITE_SUPABASE_ANON_KEY to .env file");
+}
+
+// ===== FAVORITES MANAGEMENT =====
+// Check if a quote is favorited
+async function checkIfFavorited(quoteId) {
+  if (useMockData || !supabase) {
+    // In mock mode, use localStorage
+    const favorites = JSON.parse(localStorage.getItem('mock_favorites') || '[]');
+    return favorites.includes(quoteId);
+  }
+
+  try {
+    const userId = getUserId();
+    // Try RPC function first (cast to correct types)
+    const { data, error } = await supabase
+      .rpc('is_quote_favorited', { 
+        p_user_id: userId,
+        p_quote_id: quoteId
+      });
+
+    if (error) {
+      console.error('Error checking favorite (using fallback):', error);
+      // Fallback: check localStorage
```

Comment:
Nice separation between Supabase and mock logic.
One edge case to consider: if the RPC call fails while Supabase is still online, the code silently falls back to localStorage. This could lead to a mismatch between the database and the UI state.
You might want to wrap the fallback logic in a small helper that handles re-sync once the connection is restored, or surface a warning/disable favorite actions until reconnection.